src/js/media-container.js

```diff
@@ -477,98 +478,116 @@ class MediaContainer extends globalThis.HTMLElement {
     // This allows things like autoplay and programmatic playing to also initiate hiding controls (CJP)
     this.setAttribute(Attributes.USER_INACTIVE, '');
 
-    const setInactive = () => {
-      if (this.autohide < 0) return;
-      if (this.hasAttribute(Attributes.USER_INACTIVE)) return;
+    this.addEventListener('pointerdown', this);
+    this.addEventListener('pointermove', this);
+    this.addEventListener('pointerup', this);
+    this.addEventListener('mouseleave', this);
+    this.addEventListener('keyup', this);
 
-      this.setAttribute(Attributes.USER_INACTIVE, '');
+    globalThis.window?.addEventListener('mouseup', () => {
+      this.removeAttribute(Attributes.KEYBOARD_CONTROL);
+    });
+  }
 
-      const evt = new globalThis.CustomEvent(
-        MediaStateChangeEvents.USER_INACTIVE,
-        { composed: true, bubbles: true, detail: true }
-      );
-      this.dispatchEvent(evt);
-    };
+  handleEvent(event) {
+    switch (event.type) {
+      case 'pointerdown':
+        this.#pointerDownTimeStamp = event.timeStamp;
+        break;
+      case 'pointermove':
+        this.#handlePointerMove(event);
+        break;
+      case 'pointerup':
+        this.#handlePointerUp(event);
+        break;
+      case 'mouseleave':
+        // Immediately hide if mouse leaves the container.
+        this.#setInactive();
+        break;
+      case 'keyup':
+        // Unhide for keyboard controlling.
+        this.#scheduleInactive();
+        // Allow for focus styles only when using the keyboard to navigate.
+        this.setAttribute(Attributes.KEYBOARD_CONTROL, '');
+        break;
+    }
+  }
 
-    const setActive = () => {
-      if (!this.hasAttribute(Attributes.USER_INACTIVE)) return;
+  #handlePointerMove(event) {
+    if (event.pointerType !== 'mouse') {
+      // On mobile we toggle the controls on a tap which is handled in pointerup,
+      // but Android fires pointermove events even when the user is just tapping.
+      // Prevent calling setActive() on tap because it will mess with the toggle logic.
+      const MAX_TAP_DURATION = 200;
```

Comment:
per out of band convo:
```suggestion
      const MAX_TAP_DURATION = 250;
```